cub/test/thread_reduce/catch2_test_thread_reduce_partial.cu

```diff
@@ -0,0 +1,611 @@
+ **********************************************************************************************************************/
+
+template <typename T, typename>
+struct cub_operator_to_std;
+
+template <typename T>
+struct cub_operator_to_std<T, ::cuda::std::plus<>>
+{
+  using type = ::std::plus<T>; // T: MSVC complains about possible loss of data
+};
+
+template <typename T>
+struct cub_operator_to_std<T, ::cuda::std::multiplies<>>
+{
+  using type = ::std::multiplies<T>; // T: MSVC complains about possible loss of data
+};
+
+template <typename T>
+struct cub_operator_to_std<T, ::cuda::std::bit_and<>>
+{
+  using type = ::std::bit_and<T>; // T: MSVC complains about possible loss of data
+};
+
+template <typename T>
+struct cub_operator_to_std<T, ::cuda::std::bit_or<>>
+{
+  using type = ::std::bit_or<T>; // T: MSVC complains about possible loss of data
+};
+
+template <typename T>
+struct cub_operator_to_std<T, ::cuda::std::bit_xor<>>
+{
+  using type = ::std::bit_xor<T>; // T: MSVC complains about possible loss of data
+};
+
+template <typename T>
+struct cub_operator_to_std<T, ::cuda::minimum<>>
+{
+  using type = ::cuda::minimum<>;
+};
+
+template <typename T>
+struct cub_operator_to_std<T, ::cuda::maximum<>>
+{
+  using type = ::cuda::maximum<>;
+};
+
+template <typename T, typename Operator>
+using cub_operator_to_std_t = typename cub_operator_to_std<T, Operator>::type;
+
+/***********************************************************************************************************************
+ * CUB operator to identity
+ **********************************************************************************************************************/
+
+template <typename T, typename Operator, typename = void>
+struct cub_operator_to_identity;
+
+template <typename T>
+struct cub_operator_to_identity<T, ::cuda::std::plus<>>
+{
+  static constexpr T value()
+  {
+    return T{};
+  }
+};
+
+template <typename T>
+struct cub_operator_to_identity<T, ::cuda::std::multiplies<>>
+{
+  static constexpr T value()
+  {
+    return T{1};
+  }
+};
+
+template <typename T>
+struct cub_operator_to_identity<T, ::cuda::std::bit_and<>>
+{
+  static constexpr T value()
+  {
+    return static_cast<T>(~T{0});
+  }
+};
+
+template <typename T>
+struct cub_operator_to_identity<T, ::cuda::std::bit_or<>>
+{
+  static constexpr T value()
+  {
+    return T{0};
+  }
+};
+
+template <typename T>
+struct cub_operator_to_identity<T, ::cuda::std::bit_xor<>>
+{
+  static constexpr T value()
+  {
+    return T{0};
+  }
+};
+
+template <typename T>
+struct cub_operator_to_identity<T, ::cuda::minimum<>>
+{
+  static constexpr T value()
+  {
+    return ::cuda::std::numeric_limits<T>::max();
+  }
+};
+
+template <typename T>
+struct cub_operator_to_identity<T, ::cuda::maximum<>>
+{
+  static constexpr T value()
+  {
+    return ::cuda::std::numeric_limits<T>::lowest();
+  }
+};
+
+namespace detail
+{
+
+template <typename T, typename Operator, typename = void>
+struct dist_interval
```

Comment:
This helper code is duplicated between this test and the `ThreadScan*Partial` tests. I propose to put it in a header.